provider/datagen/src/baked_exporter.rs

```diff
@@ -27,8 +68,37 @@ macro_rules! move_out {
 // TokenStream isn't Send/Sync
 type SyncTokenStream = String;
 
+/// Options for configuring the output of [`BakedExporter`].
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct Options {
+    /// Whether to run `rustfmt` on the generated files.
+    pub pretty: bool,
+    /// Whether to gate each key on its crate name. This allows using the module
+    /// even if some keys are not required and their dependencies are not included.
+    /// Requires use_separate_crates.
+    pub insert_feature_gates: bool,
+    /// Whether to use separate crates to name types instead of the `icu` metacrate
+    pub use_separate_crates: bool,
```

Comment:
Suggestion (optional): when two booleans depend on each other like this, it sometimes means that you should be using a 3-element enum instead

```rust
pub enum CrateMode {
  MetaCrate,
  SeparateCratesNoFeatures,
  SeparateCratesWithFeatures,
}
```

Reply:
This is pre-existing code (`crate::BakedOptions`) and is re-exported as that for stability.

Also, you wrote this 😀